clippy_lints/src/misc.rs

```diff
@@ -224,7 +244,23 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                                                initref=initref));
                 }
             );
-        }}
+        }};
+        if_let_chain! {[
+            let StmtSemi(ref expr, _) = s.node,
+            let Expr_::ExprBinary(ref binop, ref a, ref b) = expr.node,
+            binop.node == BiAnd || binop.node == BiOr,
+            let Some(sugg) = Sugg::hir_opt(cx, a),
```

Comment:
@oli-obk `hir_opt`/`hir` correspond to `snippet_opt`/`snippet`. `snippet_opt` can only return `None` on invalid spans and maybe if macros are involved I believe.